docs/src/main/sphinx/installation/query-resiliency.rst

```diff
@@ -0,0 +1,119 @@
+======================================================
+Improve query resiliency with fault-tolerant execution
+======================================================
+
+You can configure Trino to be more resilient against query failure
+by enabling fault-tolerant execution. This allows Trino to handle
+larger queries such as batch operations without worker node interruptions
+causing the query to fail.
+
+When configured, the Trino cluster buffers data used by workers
+during query processing. If a worker node fails for any reason, such as a
+network outage or running out of available resources, the coordinator issues the
+data to an available worker that can pick up query processing, allowing the
+query to continue.
+
+Architecture
+------------
+
+The coordinator node uses a configured exchange manager service that buffers
+data for query processing in an external location such as an S3 bucket. Worker
```

Comment:
data during query processing in an external location, such as an S3 object storage location.